course/viewbase.py

```diff
@@ -98,6 +99,11 @@ def get_resource_objects(self):
                         language = query_language
                         if languages:
                             self.user_language = languages[0]
+                            # Since the option to continue browsing the course in
+                            # enrollment language is presented in the case the user navigates
+                            # the course with a secondary language, it makes sense to set the
+                            # self.url_secondary_query_language to the user_language
+                            self.url_secondary_query_language = switch_lang_query_param_from_url(self.request.get_full_path(), 'hl', languages[0])
```

Comment:
I don't understand yet why you are adding these secondary language parameters. I would prefer a simple solution instead of adding more hacks. We can talk in more detail soon.

Reply:
It is important to handle the language query parameters in the POST requests. Therefore, we should always redirect the user to the corresponding URL and add the query parameter. Otherwise, the query parameter is deleted, and the URL to which the user is redirected will not include the `hl` parameter.  Currently, we are handling the navigation through the course, assuming that everything is a GET request (basically, that was the reason to add the JS event handler), but we should also think about the management of the POST request. In this PR, I added a third commit that shows an example of how a POST request generated when the user tries to create a group can break the management of the `hl` parameter in the URL.